BUG:Fix incorrect 'inner' method type annotation in __array_ufunc_

zachbrugh opened the pull request:
BUG:Fix incorrect 'inner' method type annotation in __array_ufunc_

Replaced the 'inner' method type annotation with 'at' in the __array_ufunc__ method signature. The 'at' method is a valid universal function method, whereas 'inner' is not recognized by NumPy's ufunc. This change corrects the static type checking issue reported in issue #25499, where using 'at' would incorrectly fail type inspection, and 'inner' would erroneously pass.

Related to issue #25499

<!--         ----------------------------------------------------------------
                MAKE SURE YOUR PR GETS THE ATTENTION IT DESERVES!
                ----------------------------------------------------------------

*  FORMAT IT RIGHT:
      https://www.numpy.org/devdocs/dev/development_workflow.html#writing-the-commit-message

*  IF IT'S A NEW FEATURE OR API CHANGE, TEST THE WATERS:
      https://www.numpy.org/devdocs/dev/development_workflow.html#get-the-mailing-list-s-opinion

*  HIT ALL THE GUIDELINES:
      https://numpy.org/devdocs/dev/index.html#guidelines

*  WHAT TO DO IF WE HAVEN'T GOTTEN BACK TO YOU:
      https://www.numpy.org/devdocs/dev/development_workflow.html#getting-your-pr-reviewed
-->

zachbrugh committed:
BUG:Fix incorrect 'inner' method type annotation in __array_ufunc_
_
Replaced the 'inner' method type annotation with 'at' in the __array_ufunc__ method signature. The 'at' method is a valid universal function method, whereas 'inner' is not recognized by NumPy's ufunc. This change corrects the static type checking issue reported in issue #25499, where using 'at' would incorrectly fail type inspection, and 'inner' would erroneously pass.

Related to issue #25499

BvB93 reviewed (commented):
Looks good, though there are two more places where this needs fixing:
* `NDArrayOperatorsMixin.__array_ufunc__` in numpy/lib/mixins.pyi
* `_SupportsArrayUFunc.__array_ufunc__` in numpy/_typing/_ufunc.pyi

zachbrugh committed:
BUG:Fix incorrect 'inner' method type annotation in NDArrayOperatorsMixin.__array_ufunc__ and _SupportsArrayUFunc.__array_ufunc__  #25510

Replaced the 'inner' method type annotation with 'at' in the  NDArrayOperatorsMixin.__array_ufunc__ and _SupportsArrayUFunc.__array_ufunc__  method signatures. The 'at' method is a valid universal function method, whereas 'inner' is not recognized by NumPy's ufunc. This change corrects the static type checking issue reported in issue #25499, where using 'at' would incorrectly fail type inspection, and 'inner' would erroneously pass.

zachbrugh committed:
Merge branch 'numpy:main' into type-annotation-ndarray.__array_ufunc__

BvB93 commented:
LGTM, thanks @zachbrugh
